### PR TITLE
[HFH-1733] Lighter/safer SCIM information

### PR DESCRIPTION
**Pull request by xjunior**
Exposes less data from SCIM resources through the proxy, limiting to the information that we care about. This also affects how the information is cached within the context criteria, reducing the size of the stored JSON.

- Safer SCIM proxy exposing only the necessary
- Avoid unnecessary request to SCIM discovery endpoint

### CODE REVIEW

File: audiences-react/src/AudienceForm/CriteriaListFields.tsx [hunk 1]
```diff
@@ -8,19 +8,18 @@ import omitBy from "lodash/omitBy"
...
-import { ScimResourceType } from "../useScimResources"
 import { GroupCriteria } from "../types"
 
 type GroupCriteriaField = GroupCriteria & {
   id: string
 }
 type CriteriaListFieldsProps = {
   name: string
-  resources: ScimResourceType[]
+  groupResources: string[]
```
Comment (that-jill) [inline audiences-react/src/AudienceForm/CriteriaListFields.tsx:18]: is there no way to have a cleaner type on here?

File: audiences-react/src/AudienceForm/ScimResourceTypeahead.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import debounce from "lodash/debounce"
 import { useController } from "react-hook-form"
+import get from "lodash/get"
```
Comment (that-jill) [inline audiences-react/src/AudienceForm/ScimResourceTypeahead.tsx:3]: are we using this?

File: audiences-react/src/AudienceForm/CriteriaListFields.tsx (hunk 1 above)
Comment (xjunior) [reply audiences-react/src/AudienceForm/CriteriaListFields.tsx:18]: @that-jill it really is a variable string. It was the whole [ResourceType information](https://developer.4me.com/v1/scim/resource_types/) before (name, id, endpoint, etc), but that was unnecessary.

File: audiences-react/src/AudienceForm/ScimResourceTypeahead.tsx (hunk 2 above)
Comment (xjunior) [reply audiences-react/src/AudienceForm/ScimResourceTypeahead.tsx:3]: Yep, line 20

Comment (that-jill) [reply audiences-react/src/AudienceForm/ScimResourceTypeahead.tsx:3]: Weird that we were using it without the require before but okay

**Review by that-jill — APPROVED** [review]

**Pull request merged**